docs/tutorials/text_sentiment_ngrams.ipynb

```diff
@@ -0,0 +1,496 @@
+    "class TextSentiment(nn.Module):\n",
+    "    def __init__(self, vocab_size, embed_dim, num_class):\n",
+    "        super().__init__()\n",
+    "        self.embedding = nn.EmbeddingBag(vocab_size, embed_dim, sparse=True)\n",
+    "        self.fc = nn.Linear(embed_dim, num_class)\n",
+    "        self.init_weights()\n",
+    "\n",
+    "    def init_weights(self):\n",
+    "        initrange = 0.5\n",
+    "        self.embedding.weight.data.uniform_(-initrange, initrange)\n",
+    "        self.fc.weight.data.uniform_(-initrange, initrange)\n",
+    "        self.fc.bias.data.zero_()\n",
+    "        \n",
+    "    def forward(self, text, offsets):\n",
+    "        embedded = self.embedding(text, offsets)\n",
+    "        return self.fc(embedded)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Initiate an instance\n",
+    "\n",
+    "The AG_NEWS dataset has four labels and therefore the number of classes is four.\n",
+    "\n",
+    "    1 : World\n",
+    "    2 : Sports\n",
+    "    3 : Business\n",
+    "    4 : Sci/Tec\n",
+    "\n",
+    "The vocab size is equal to the length of vocab (including single word and ngrams). The number of classes is equal to the number of labels, which is four in AG_NEWS case."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "VOCAB_SIZE = len(train_dataset.get_vocab())\n",
+    "EMBED_DIM = 512\n",
+    "NUN_CLASS = len(train_dataset.get_labels())\n",
+    "model = TextSentiment(VOCAB_SIZE, EMBED_DIM, NUN_CLASS).to(device)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "lines_to_next_cell": 2
+   },
+   "source": [
+    "## Add optimizer and loss function\n",
+    "\n",
+    "[SGD](https://pytorch.org/docs/stable/optim.html?highlight=sgd#torch.optim.SGD) algorithm is used to optimize the model. \n",
+    "[CrossEntropyLoss](https://pytorch.org/docs/stable/nn.html?highlight=crossentropyloss#torch.nn.CrossEntropyLoss) criterion combines nn.LogSoftmax() and nn.NLLLoss() in one single class. It is useful when training a classification problem with C classes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import torch.optim as optim\n",
+    "optimizer = optim.SGD(model.parameters(), lr=4.0)\n",
+    "loss_func = torch.nn.CrossEntropyLoss().to(device)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Functions used to generate batch"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Functions generate_batch() is used to generate data batches and offsets, which are compatible with EmbeddingBag. EmbeddingBad is efficient to process a sequence of tensors with different lengths so that no padding is required here. This is faster than the origin method depending on TorchText Iterator and Batch."
```

Comment:
I think you mean "EmbeddingBag" :)

Reply:
Can you expand a bit on the "original method depending on TorchText Iterator" and why it's faster?